### PR TITLE
Migrate `delete_or_timeout` function to bot class

### PR DESCRIPTION
**Pull request by ALazyMeme**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable
- [x] Documentation in docs/ or install-docs/ was updated, if applicable
- [ ] I have tested all changes

### CODE REVIEW

File: pajbot/bot.py [hunk 1]
```diff
@@ -664,6 +664,32 @@ def timeout_warn(
...
+        reason: Optional[str] = None,
+        disable_warnings: bool = False,
+        once: bool = False,
+    ) -> None:
+        if moderation_action not in ("Delete", "Timeout"):
+            log.warning("moderation_action must only equal Delete or Timeout.")
+            return
+
+        if moderation_action == "Delete" and msg_id is None:
+            log.warning("Cannot use the Delete moderation_action if msg_id is None!!!! BabyRage")
```
Comment (RAnders00) [inline pajbot/bot.py:682]: lol. But I'd rather you throw an exception here

File: pajbot/bot.py [hunk 2]
```diff
@@ -664,6 +664,32 @@ def timeout_warn(
...
+            return
+
+        if moderation_action == "Delete":
+            self.delete_message(msg_id)
+        elif moderation_action == "Timeout":
+            if disable_warnings is True:
```
Comment (RAnders00) [inline pajbot/bot.py:688]:
```suggestion
            if disable_warnings:
```

File: pajbot/bot.py [hunk 3]
```diff
@@ -664,6 +664,32 @@ def timeout_warn(
...
+        user: User,
+        moderation_action: str,
+        msg_id: str,
+        duration: int,
+        reason: Optional[str] = None,
+        disable_warnings: bool = False,
+        once: bool = False,
+    ) -> None:
+        if moderation_action not in ("Delete", "Timeout"):
+            log.warning("moderation_action must only equal Delete or Timeout.")
```
Comment (RAnders00) [inline pajbot/bot.py:678]: I'd rather you throw an exception here

**Review by RAnders00 — CHANGES_REQUESTED** [review]

**Comment by ALazyMeme** [comment]
Will test once initial code is approved.

File: pajbot/bot.py [hunk 4]
```diff
@@ -664,6 +664,32 @@ def timeout_warn(
...
+        moderation_action: str,
+        msg_id: str,
+        duration: int,
+        reason: Optional[str] = None,
+        disable_warnings: bool = False,
+        once: bool = False,
+    ) -> None:
+        if moderation_action not in ("Delete", "Timeout"):
+            log.exception("moderation_action must only equal Delete or Timeout!")
+            return
```
Comment (RAnders00) [inline pajbot/bot.py:679]:
```suggestion
            raise ValueError("moderation_action must only equal Delete or Timeout!")
```

File: pajbot/bot.py [hunk 5]
```diff
@@ -664,6 +664,32 @@ def timeout_warn(
...
+
+        if moderation_action == "Delete" and msg_id is None:
+            log.exception("Cannot use the Delete moderation_action if msg_id is None!")
+            return
```
Comment (RAnders00) [inline pajbot/bot.py:683]:
```suggestion
            raise ValueError("Cannot use the Delete moderation_action if msg_id is None!")
```

**Review by RAnders00 — CHANGES_REQUESTED** [review]
I was thinking like this

**Comment by RAnders00** [comment]
I‘d like to get @pajlada‘s opinion on this, I‘m not a super huge fan but it‘s alrighg

**Comment by ALazyMeme** [comment]
> I‘m not a super huge fan

Why's that? The idea is mostly just to centralize the function i/o creating it in every module we want to use it in.

**Review by pajlada — APPROVED** [review]

**Pull request merged**